standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1396,9 +1400,11 @@ public AllocateTableWriteIdsResponse allocateTableWriteIds(AllocateTableWriteIds
           }
           txnIds = getTargetTxnIdList(rqst.getReplPolicy(), srcTxnIds, stmt);
           if (srcTxnIds.size() != txnIds.size()) {
-            LOG.warn("Target txn id is missing for source txn id : " + srcTxnIds.toString() +
+            // Idempotent case where txn was already closed but gets allocate write id event.
```

Comment:
it should be either all or none case ..

Reply:
not necessarily all or none. If allocatewriteId event comes with batch of txns but abort/commit event comes individually. If few of aborts are applied means, here few txns may exist. But, we shouldn't allow allocating write ids as it would've been already applied or abort would follow.